File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/SpecificationController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.sellergoods.service.SpecificationService;
import entity.PageResult;
import entity.Result;
import entityGroup.Specification;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("specification")
public class SpecificationController {

    @Reference
    private SpecificationService service;

    @RequestMapping("findSpecList")
    public List<Map> findSpecList() {
        return service.findSpecList();
    }


    @RequestMapping("findAll")
    public List<TbSpecification> findAll() {
        return service.findAll();
    }

    @RequestMapping("findPage/{pageNum}/{pageSize}")
    public PageResult findPage(@PathVariable("pageNum") int pageNum, @PathVariable("pageSize") int pageSize) {
        return service.findPage(pageNum, pageSize);
    }

    @RequestMapping("search/{pageNum}/{pageSize}")
    public PageResult search(@PathVariable("pageNum") int pageNum, @PathVariable("pageSize") int pageSize, @RequestBody TbSpecification specification) {
        return service.search(pageNum, pageSize, specification);
    }

    @RequestMapping("add")
    public Result add(@RequestBody Specification specification) {
        try {
            service.add(specification);
            return new Result(true, "添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "添加失败");
        }
    }

    @RequestMapping("findOne/{id}")
    public Specification findOne(@PathVariable("id") Long id) {
        return service.findOne(id);
    }

    @RequestMapping("update")
    public Result update(@RequestBody Specification specification) {
        try {
            service.update(specification);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    @RequestMapping("del/{ids}")
    public Result del(@PathVariable Long[] ids) {
        try {
            service.del(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }
    }
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/specificationController.js
app.controller("specificationController", function ($scope, $controller, specificationService) {//依赖注入service

    $controller("baseController",{$scope:$scope});//继承baseController控制器

    //初始化entity
    $scope.entity={specificationOptionList:[]};

    //动态添加规格项
    $scope.addSpecification=function () {
        $scope.entity.specificationOptionList.push({});
    };

    //动态删除规格项
    $scope.delSpecification=function (index) {
        $scope.entity.specificationOptionList.splice(index,1);
    };

    //分页显示
    $scope.findPage = function (pageNum, pageSize) {
        specificationService.findPage(pageNum, pageSize).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    };

    //条件查询
    $scope.search = function (pageNum, pageSize) {
        specificationService.search(pageNum, pageSize, $scope.searchEntity).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    };

    //添加品牌
    $scope.save = function () {
        var resultObj = null;
        if ($scope.entity.specification.id != null) {
            resultObj = specificationService.update($scope.entity);
        } else {
            resultObj = specificationService.add($scope.entity);
        }
        resultObj.success(function (response) {
            //需要的response： {"true"|"false":"添加成功"|"添加失败"}
            if (response.success) {
                $scope.reloadList();
            } else {
                alert(response.message);
            }
        })
    };

    //修改品牌
    $scope.findOne = function (id) {
        specificationService.findOne(id).success(function (response) {
            //将response赋值给entity来回显数据
            $scope.entity = response;
        })
    };

    //删除品牌
    $scope.del = function () {
        if ($scope.Ids.length > 0) {
            if (window.confirm("确定要删除吗？")) {
                specificationService.del($scope.Ids).success(function (response) {
                    if (response.success) {
                        $scope.reloadList();
                        $scope.Ids = [];
                    } else {
                        alert(response.message);
                    }
                })
            }
        }
    };

//查询所有
    $scope.findAll = function () {
        specificationService.findAll().success(function (response) {
            $scope.list = response;
        })
    };

});

